app/ui.js

```diff
@@ -235,6 +236,27 @@ var UI;
             UI.initSetting('reconnect_delay', 5000);
         },
 
+        setupSettingLabels: function() {
+            var labels = document.getElementsByTagName('LABEL');
+            for (var i = 0; i < labels.length; i++) {
+                var label = labels[i];
+                var htmlFor = label.htmlFor;
+                if (htmlFor != '') {
+                    var elem = document.getElementById(htmlFor);
+                    if (elem) elem.label = label;
+                } else {
+                    // If 'for' isn't set, let's use the first non-text child
```

Comment:
Shouldn't you look for input like elements rather than just anything? I.e. check if `Element.form !== undefined`.